packages/react-components/src/number-input/src/NumberInput.tsx

```diff
@@ -1,89 +1,111 @@
 import { useMemo } from "react";
 import { useToolbarProps } from "../../toolbar";
 
-const propTypes = {
+export interface InnerNumberInputProps extends InteractionStatesProps {
+    /**
+     * @ignore
+     */
+    id?: string;
     /**
      * A controlled value.
      */
-    value: number,
+    value?: number,
     /**
      * The default value of `value` when uncontrolled.
      */
-    defaultValue: number,
+    defaultValue?: number,
+    /**
+     * Whether or not the input is readonly.
+     */
+    readOnly?: boolean;
+    /**
+     * Whether or not a user input is required before form submission.
+     */
+    required?: boolean;
     /**
      * Temporary text that occupies the input when it is empty.
      */
-    placeholder: string,
+    placeholder?: string,
     /**
      * The minimum value of the input. See [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number).
      */
-    min: number,
+    min?: number,
     /**
      * The maximum value of the input. See [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number).
      */
-    max: number,
+    max?: number,
     /**
      * The step used to increment or decrement the value. See [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number).
      */
-    step: number,
+    step?: number,
     /**
      * Clamps the input value between min & max boundaries.
      */
-    clampValue: bool,
+    clampValue?: boolean;
     /**
      * Whether or not the input should display as "valid" or "invalid".
      */
-    validationState: oneOf(["valid", "invalid"]),
+    validationState?: "valid" | "invalid";
     /**
      * Called when the input value change.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @param {number} value - The new value.
      * @returns {void}
      */
-    onChange: func,
+    onChange?(event: SyntheticEvent, value: number): void;
     /**
      * Whether or not the input should autofocus on render.
      */
-    autoFocus: oneOfType([bool, number]),
+    autoFocus?: boolean | number;
     /**
      * [Icon](/?path=/docs/icon--default-story) component rendered before the value.
      */
-    icon: element,
+    icon?: ReactElement,
     /**
      * Whether or not the input take up the width of its container.
      */
-    fluid: bool,
+    fluid?: boolean;
     /**
      * Whether or not to render a loader.
      */
-    loading: bool,
+    loading?: boolean;
     /**
      * Additional props to render on the wrapper element.
      */
-    wrapperProps: object,
+    wrapperProps?: Partial<BoxProps>,
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    as: oneOfType([string, elementType])
-};
+    as?: ElementType
+    /**
+    * @ignore
+    */
+    forwardedRef: ForwardedRef<any>;
+}
+
+export interface SpinnerProps extends ComponentProps<"div"> {
```

Comment:
Spinner is exported as Spinner, not NumberInputSpinner. Why would the itnerface name be different? should we not export Spinner?

Reply:
Well Spinner doesn't mean anything if it's not contextual to a NumberInput. It could be expected to be a lot of things otherwise. For example, a loading spinner.